lib/active_merchant/billing/gateways/merchant_partners.rb

```diff
@@ -0,0 +1,253 @@
+        transcript.
+          gsub(%r((<ccnum>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<cvv2>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<merchantpin>)[^<]+(<))i, '\1[FILTERED]\2')
+      end
+
+      def test?
+        @options[:account_id].eql?('TEST0')
+      end
+
+      private
+
+      def add_invoice(post, money, options)
+        post[:amount] = amount(money)
+        post[:merchantordernumber] = options[:order_id]
+        post[:currency] = options[:currency] || currency(money)
+      end
+
+      def add_payment_method(post, payment_method)
+        if(payment_method.is_a?(String))
+          user_profile_id, last_4 = split_authorization(payment_method)
+          post[:userprofileid] = user_profile_id
+          post[:last4digits] = last_4
+        else
+          post[:ccname] = payment_method.name
+          post[:ccnum] = payment_method.number
+          post[:cvv2] = payment_method.verification_value
+          post[:expmon] = format(payment_method.month, :two_digits)
+          post[:expyear] = format(payment_method.year, :four_digits)
+          post[:swipedata] = payment_method.track_data if payment_method.track_data
+        end
+      end
+
+      def add_customer_data(post, options)
+        post[:email] = options[:email] if options[:email]
+        post[:ipaddress] = options[:ip] if options[:ip]
+        if(billing_address = (options[:billing_address] || options[:address]))
+          post[:billaddr1] = billing_address[:address1]
+          post[:billaddr2] = billing_address[:address2]
+          post[:billcity] = billing_address[:city]
+          post[:billstate] = billing_address[:state]
+          post[:billcountry] = billing_address[:country]
+          post[:bilzip]    = billing_address[:zip]
+          post[:phone] = billing_address[:phone]
+        end
+      end
+
+      def add_reference(post, authorization)
+        post[:historykeyid] = authorization
+      end
+
+      ACTIONS = {
+        "purchase" => "2",
+        "authorize" => "1",
+        "capture" => "3",
+        "void" => "5",
+        "refund" => "4",
+        "credit" => "6",
+        "store" => "7",
+        "stored_purchase" => '8',
+        "stored_credit" => '13'
+      }
+
+      def commit(action, post)
+        post[:acctid] = @options[:account_id]
+        post[:merchantpin] = @options[:merchant_pin]
+        post[:service] = ACTIONS[action] if ACTIONS[action]
+
+        data = build_request(post)
+        response_data = parse(ssl_post(url(action), data, headers))
+        succeeded = success_from(response_data)
+
+        Response.new(
+          succeeded,
+          message_from(succeeded, response_data),
+          response_data,
+          authorization: authorization_from(post, response_data),
+          :avs_result => AVSResult.new(code: response_data["avs_response"]),
+          :cvv_result => CVVResult.new(response_data["cvv2_response"]),
+          error_code: error_code_from(succeeded, response_data),
```

Comment:
I'm thinking this line ought to be removed.  The `#error_code_from` could be removed as well.  My understanding is that `error_code` is used when you've got a real mapping to STANDARD_ERROR_CODE which we won't have for this gateway for now.